### PR TITLE
Added 7zip support thru py7zr.

### PR DESCRIPTION
**Pull request by MichaelFitzurka**
Tweaked save of archive file and images in comicarchive.

### CODE REVIEW

File: scripts/name_fixer.py [hunk 1]
```diff
@@ -54,7 +54,9 @@ def calculate_rename(ca, md, settings):
 
     new_ext = None  # default
     if settings.rename_extension_based_on_archive:
-        if ca.isZip():
+        if ca.isSevenZip():
+            new_ext = ".cb7"
+        elif ca.isZip():
```
Comment (lordwelch) [inline scripts/name_fixer.py]: Let's not edit the scripts unless we are also going to update them to python3, it makes it simpler to trace their history when they aren't muddled by ineffective edits.

File: scripts/remove_ads.py [hunk 2]
```diff
@@ -51,7 +51,7 @@ def main():
     for filename in filelist:
 
         ca = ComicArchive(filename, settings.rar_exe_path)
-        if (ca.isZip or ca.isRar()) and ca.hasMetadata(style):
+        if (ca.isSevenZip() or ca.isZip() or ca.isRar()) and ca.hasMetadata(style):
```
Comment (lordwelch) [inline scripts/remove_ads.py]: Same as above

File: comicapi/comicarchive.py [hunk 3]
```diff
@@ -1037,7 +1153,10 @@ def applyArchiveInfoToMetadata(self, md, calc_page_sizes=False):
                         data = self.getPage(idx)
                         if data is not None:
                             try:
-                                im = Image.open(io.StringIO(data))
+                                if isinstance(data, bytes):
+                                    im = Image.open(io.BytesIO(data))
+                                else:
+                                    im = Image.open(io.StringIO(data))
```
Comment (lordwelch) [inline comicapi/comicarchive.py]: What does this have to do with 7z?

File: comicapi/comicarchive.py [hunk 4]
```diff
@@ -50,6 +51,109 @@ class MetaDataStyle:
...
+    def setArchiveComment(self, comment):
+        return False
+
+    def readArchiveFile(self, archive_file):
+        data = ""
+        try:
+            with py7zr.SevenZipFile(self.path, 'r') as zf:
+                data = zf.read(archive_file)[archive_file].read()
+        except py7zr.Bad7zFile as e:
+            print("bad zipfile [{0}]: {1} :: {2}".format(e, self.path,
```
Comment (lordwelch) [inline comicapi/comicarchive.py]: This should reference 7zip and not zipfile

File: comicapi/comicarchive.py [hunk 5]
```diff
@@ -50,6 +51,109 @@ class MetaDataStyle:
...
+                archive_file), file=sys.stderr)
+            raise IOError
+        except Exception as e:
+            print("bad zipfile [{0}]: {1} :: {2}".format(e, self.path,
```
Comment (lordwelch) [inline comicapi/comicarchive.py]: same as above

**Review by lordwelch — CHANGES_REQUESTED** [review]

File: comicapi/comicarchive.py (hunk 4 above)
Comment (MichaelFitzurka) [reply comicapi/comicarchive.py]: copy past error - I'll fix

File: comicapi/comicarchive.py (hunk 3 above)
Comment (MichaelFitzurka) [reply comicapi/comicarchive.py]: I can't remember why I made this change - i think it had to do with  a png file, but am not 100%.
I am 100% sure it wasn't 7zip - so I'll remove

File: comicapi/comicarchive.py (hunk 5 above)
Comment (MichaelFitzurka) [reply comicapi/comicarchive.py]: CPE

**Review by lordwelch — APPROVED** [review]
